api/v1alpha1/openstackserver_types.go

```diff
@@ -48,8 +48,13 @@ type OpenStackServerSpec struct {
 	ConfigDrive optional.Bool `json:"configDrive,omitempty"`
 
 	// The flavor reference for the flavor for the server instance.
-	// +required
-	Flavor string `json:"flavor"`
+	// +optional
+	Flavor string `json:"flavor,omitempty"`
```

Comment:
I think we can do something like:

```
+kubebuilder:validation:XValidation:message="at least one of flavor and flavorID must be set",rule=(has(self.flavor) || has(self.flavorID))
```

on the Spec struct to tighten up the UX...

Reply:
Please can you add the above CEL validation both here and on OpenStackMachineSpec, and also add tests for it in `apivalidations`. There are lots of examples in there and they're really easy to write, so shouldn't take long.

Please can you make this a `*string` and give it a MinLength of 1, and the same for FlavorID. This prevents the empty string from being valid, whilst still allowing it to be optional. And the same for 3 versions of OpenStackMachine.